src/main.rs

```diff
@@ -858,6 +859,21 @@ mod tests {
         Parser::parse(contents.into().as_bytes()).unwrap()
     }
 
+    // Parse a string of code that is meant to fail
+    fn parse_str_fail(contents: impl Into<String>) -> Parser {
+        // Parse the code
+        let res = Parser::parse(contents.into().as_bytes());
+        // If the code parsed without error
+        if res.is_ok() {
```

Comment:
I think it's better to use `if let Ok(...) = .. ` here because then we use the variable inside the `Ok()` to printout, it's just a bit less bloated in the debug output because now it will also print Ok in the debug output.

```suggestion
        if let Ok(parsed_content) = res {
```